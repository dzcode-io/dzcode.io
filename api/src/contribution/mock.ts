import { ContributionEntity, FilterEntity, OptionEntity } from "@dzcode.io/common/dist/types";
import { lorem } from "faker";

export const generateContributionMock = (index: number): ContributionEntity => {
  const projectName = `${lorem.word()}/${lorem.word()}`;

  return {
    id: `${index}`,
    labels: lorem.sentence().split(" "),
    languages: lorem.sentence().split(" "),
    project: {
      slug: `github/${projectName}`,
      name: projectName,
    },
    title: lorem.sentence(),
    type: index % 2 ? "pullRequest" : "issue",
    url: `https://github.com/${projectName}/issues/${index}`,
    createdAt: new Date().toISOString(),
    updatedAt: new Date().toISOString(),
    commentsCount: Math.round(Math.random() * 50),
  };
};

export const bulkGenerateContributionMock = (from: number, to: number) => {
  const contributionsMock: ContributionEntity[] = [];
  const filtersMock: FilterEntity[] = [
    { label: "Project", name: "projects", options: [] },
    { label: "Language", name: "languages", options: [] },
    { label: "Label", name: "labels", options: [] },
  ];

  const pushUniqueOption = (options: OptionEntity[], filterOptions: OptionEntity[]) => {
    const uniqueOptions = options.filter(
      (_option) => !filterOptions.some(({ name }) => _option.name === name),
    );
    filterOptions.push(...uniqueOptions);
  };

  for (let i = from; i <= to; i++) {
    const contributionMock = generateContributionMock(i);
    contributionsMock.push(contributionMock);
    pushUniqueOption(
      [
        {
          name: contributionMock.project.slug,
          label: contributionMock.project.name,
        },
      ],
      filtersMock[0].options,
    );
    pushUniqueOption(
      contributionMock.languages.map((language) => ({
        name: language,
        label: language,
      })),
      filtersMock[1].options,
    );

    pushUniqueOption(
      contributionMock.labels.map((label) => ({
        name: label,
        label: label,
      })),
      filtersMock[2].options,
    );
  }

  return { contributions: contributionsMock, filters: filtersMock };
};
