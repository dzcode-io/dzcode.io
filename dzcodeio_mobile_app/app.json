{
  "name": "dzcodeio_mobile_app",
  "displayName": "dzcodeio_mobile_app"
}