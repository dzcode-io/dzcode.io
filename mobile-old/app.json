{
  "expo": {
    "name": "DzCode i/o",
    "slug": "dzcode",
    "version": "4.15.1",
    "orientation": "portrait",
    "icon": "./src/assets/png/icon.png",
    "scheme": "dzcode",
    "userInterfaceStyle": "automatic",
    "splash": {
      "image": "./src/assets/png/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#000"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": ["**/*"],
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "io.dzcode.mobile",
      "buildNumber": "4.15.1",
      "userInterfaceStyle": "automatic"
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./src/assets/png/adaptive-icon.png",
        "backgroundColor": "#000"
      },
      "permissions": [],
      "package": "io.dzcode.mobile",
      "versionCode": 4151,
      "userInterfaceStyle": "automatic"
    },
    "web": {
      "favicon": "./src/assets/png/favicon.png"
    },
    "plugins": ["sentry-expo"],
    "hooks": {
      "postExport": [
        {
          "file": "sentry-expo/upload-sourcemaps"
        }
      ]
    }
  }
}
