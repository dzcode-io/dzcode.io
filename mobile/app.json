{
  "expo": {
    "name": "DzCode i/o",
    "slug": "dzcode",
    "version": "1.1.2",
    "orientation": "portrait",
    "icon": "./src/assets/images/icon.png",
    "scheme": "dzcode",
    "userInterfaceStyle": "automatic",
    "splash": {
      "image": "./src/assets/images/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#000"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": ["**/*"],
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "io.dzcode.mobile",
      "buildNumber": "1.1.2"
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./src/assets/images/adaptive-icon.png",
        "backgroundColor": "#000"
      },
      "permissions": [],
      "package": "io.dzcode.mobile",
      "versionCode": 112
    },
    "web": {
      "favicon": "./src/assets/images/favicon.png"
    },
    "plugins": ["sentry-expo"]
  }
}
