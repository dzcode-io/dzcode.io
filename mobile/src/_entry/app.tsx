import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";

export const App = () => {
  return (
    <View style={styles.container}>
      <Text>moved entry</Text>
      <StatusBar style="auto" />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
