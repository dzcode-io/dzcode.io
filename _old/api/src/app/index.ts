import "reflect-metadata";

import { fsConfig } from "@dzcode.io/utils/dist/config";
import * as Sentry from "@sentry/node";
import { Application } from "express";
import { createExpressServer, RoutingControllersOptions, useContainer } from "routing-controllers";
import { ArticleController } from "src/article/controller";
import { ConfigService } from "src/config/service";
import { ContributionController } from "src/contribution/controller";
import { DocumentationController } from "src/documentation/controller";
import { GithubController } from "src/github/controller";
import { LoggerService } from "src/logger/service";
import { MilestoneController } from "src/milestone/controller";
import { ProjectController } from "src/project/controller";
import { TeamController } from "src/team/controller";
import Container from "typedi";

import { DocsMiddleware } from "./middlewares/docs";
import { ErrorMiddleware } from "./middlewares/error";
import { LoggerMiddleware } from "./middlewares/logger";
import { RobotsMiddleware } from "./middlewares/robots";
import { SecurityMiddleware } from "./middlewares/security";
import { SentryErrorHandlerMiddleware } from "./middlewares/sentry-error-handler";
import { SentryRequestHandlerMiddleware } from "./middlewares/sentry-request-handler";

// Use typedi container
useContainer(Container);

const { NODE_ENV, PORT, BUNDLE_INFO } = Container.get(ConfigService).env();

if (NODE_ENV !== "development") {
  Sentry.init({
    dsn: "https://5f9d7ae6e98944e1815f8d1944fc3c12@o953637.ingest.sentry.io/5904452",
    tracesSampleRate: 1.0,
    environment: NODE_ENV,
    debug: NODE_ENV !== "production",
    release: `api@${BUNDLE_INFO.version}`,
  });
}

// Create the app:
export const routingControllersOptions: RoutingControllersOptions = {
  controllers: [
    ContributionController,
    TeamController,
    GithubController,
    MilestoneController,
    ProjectController,
    ArticleController,
    DocumentationController,
  ],
  middlewares: [
    SentryRequestHandlerMiddleware,
    SecurityMiddleware,
    SentryErrorHandlerMiddleware,
    ErrorMiddleware,
    LoggerMiddleware,
    DocsMiddleware,
    RobotsMiddleware,
  ],
  defaultErrorHandler: false,
  cors: Container.get(SecurityMiddleware).cors(),
};
const app: Application = createExpressServer(routingControllersOptions);

const logger = Container.get(LoggerService);

// Start it
app.listen(PORT, () => {
  const commonConfig = fsConfig(NODE_ENV);
  logger.info({ message: `API Server up on: ${commonConfig.api.url}/docs` });
});
