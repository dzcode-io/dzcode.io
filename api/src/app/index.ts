import "reflect-metadata";

import { createExpressServer, useContainer } from "routing-controllers";
import { Application } from "express";
import { ConfigService } from "../config/service";
import Container from "typedi";
import { ContributionController } from "../contribution/controller";
import { ContributorController } from "../contributor/controller";
import { DocsMiddleware } from "./middlewares/docs";
import { ErrorMiddleware } from "./middlewares/error";
import { GithubUserController } from "../github-user/controller";
import { LoggerMiddleware } from "./middlewares/logger";
import { LoggerService } from "../logger/service";
import { SecurityMiddleware } from "./middlewares/security";
import { fsConfig } from "@dzcode.io/common/dist/config";
import router from "./routes/api";

// Use typedi container
useContainer(Container);

// Create the app:
export const routingControllersOptions = {
  controllers: [ContributionController, ContributorController, GithubUserController],
  middlewares: [
    // middlewares:
    SecurityMiddleware,
    ErrorMiddleware,
    LoggerMiddleware,
    DocsMiddleware,
  ],
  routePrefix: "/v2",
  defaultErrorHandler: false,
  cors: Container.get(SecurityMiddleware).cors(),
};
const app: Application = createExpressServer(routingControllersOptions);

// Load old code to the app, temporarily until we migrate all endpoints
app.use("/", router);

const { NODE_ENV, PORT } = Container.get(ConfigService).env();
const logger = Container.get(LoggerService);

// Start it
app.listen(PORT, () => {
  const commonConfig = fsConfig(NODE_ENV);
  logger.info({ message: `API Server up on: ${commonConfig.api.url}/v2/docs` });
});
