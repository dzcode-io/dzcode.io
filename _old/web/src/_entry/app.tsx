import "./style.scss";

import { allLanguages, LanguageEntity } from "@dzcode.io/models/dist/language";
import { Flex, MAX_CONTAINER_WIDTH } from "@dzcode.io/ui/dist/flex";
import { Footer } from "@dzcode.io/ui/dist/footer";
import { Loading } from "@dzcode.io/ui/dist/loading";
import { Navbar } from "@dzcode.io/ui/dist/navbar";
import { Stack } from "@dzcode.io/ui/dist/stack";
import { FC, Suspense, useEffect } from "react";
import { Helmet } from "react-helmet";
import { Route, RouteProps, Switch, useLocation, useRouteMatch } from "react-router-dom";
import logoWide from "src/assets/svg/logo-wide.svg";
import logo from "src/assets/svg/logo-wide.svg";
import logoExtended from "src/assets/svg/logo-wide-extended.svg";
import { L } from "src/components/l";
import { t } from "src/components/t";
import { actions } from "src/redux";
import { useSliceSelector } from "src/redux/selectors";
import { getEnv } from "src/utils";
import { urlLanguageRegEx } from "src/utils/language";

interface RouteInterface extends RouteProps {
  pageName: string;
}

const routes: RouteInterface[] = [
  {
    pageName: "landing",
    path: "/",
    exact: true,
  },
  {
    pageName: "learn",
    path: "/Learn/:articleId*",
  },
  {
    pageName: "projects",
    path: "/Projects",
  },
  {
    pageName: "articles",
    path: "/Articles/:articleId*",
  },
  {
    pageName: "faq",
    path: "/FAQ",
  },
  {
    pageName: "contribute",
    path: "/Contribute",
  },
  {
    pageName: "team",
    path: "/Team",
  },
  {
    pageName: "not-found",
  },
];

export const App: FC = () => {
  const location = useLocation();
  const match = useRouteMatch<{ lang?: LanguageEntity["code"] }>(urlLanguageRegEx);
  const landingPageMatch = useRouteMatch<{ lang?: LanguageEntity["code"] }>(`${urlLanguageRegEx}/`);
  const { language, themeName } = useSliceSelector("settings");
  const { links } = useSliceSelector("navbarComponent");
  const { sections } = useSliceSelector("footerComponent");

  useEffect(() => {
    if (getEnv() !== "development") {
      if (window.ga) {
        window.ga("set", "page", location.pathname);
        window.ga("send", "pageview");
      }
    }

    const urlLanguage =
      allLanguages.find(({ code }) => code === match?.params.lang) || allLanguages[0];
    if (urlLanguage.code !== language.code) {
      actions.settings.set({ language: urlLanguage });
    }
  }, [location]);

  return (
    <>
      <Helmet>
        <title>{t("landing-title")}</title>
      </Helmet>
      <Stack direction="vertical" min={{ height: "100vh" }}>
        <Navbar
          version={window.bundleInfo.version}
          selectedLanguageCode={language.code}
          themeName={themeName}
          logo={logo}
          logoExtended={logoExtended}
          links={links}
          onLanguageChanged={(selectedLanguageCode) => {
            actions.settings.set({
              language: allLanguages.find(({ code }) => code === selectedLanguageCode),
            });
          }}
          onThemeChanged={(selectedThemeName) => {
            actions.settings.set({ themeName: selectedThemeName });
          }}
          fixed={!!landingPageMatch?.isExact}
        />
        <Flex max={{ width: MAX_CONTAINER_WIDTH }} grow={1} display="flex">
          <Suspense fallback={<Loading logo={logoWide} />}>
            <Switch>
              {routes.map(({ pageName, path, ...route }, index) => (
                <Route
                  {...route}
                  path={path ? `${urlLanguageRegEx}${path}` : undefined}
                  key={`route-${index}`}
                  render={() => <L page={pageName} />}
                />
              ))}
            </Switch>
          </Suspense>
        </Flex>
        <Footer sections={sections} bottomText="footer-bottom-text" />
      </Stack>
    </>
  );
};
