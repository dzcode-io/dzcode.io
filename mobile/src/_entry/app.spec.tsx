import React, { Component } from "react";
import renderer from "react-test-renderer";

import { App } from "./app";

jest.useFakeTimers();

const mock = jest.requireMock("react-native-reanimated");
jest.mock("react-native-reanimated", () => ({
  ...mock,
  useSharedValue: jest.fn().mockReturnValue(0),
  useAnimatedStyle: jest.fn().mockReturnValue({}),
  useAnimatedScrollHandler: jest.fn().mockReturnValue({}),
  createAnimatedComponent: (component: Component) => jest.fn().mockReturnValue(component),
  __reanimatedWorkletInit: jest.fn(),
  ScrollView: "ScrollView",
}));

jest.mock("@gorhom/bottom-sheet", () => {
  require("react-native-reanimated/mock");
});

(global as any).__reanimatedWorkletInit = jest.fn(); // eslint-disable-line @typescript-eslint/no-explicit-any

jest.mock("@react-navigation/drawer", () => ({
  createDrawerNavigator: jest.fn().mockReturnValue({
    Navigator: jest.fn().mockReturnValue(null),
    Screen: jest.fn().mockReturnValue(null),
  }),
}));

it("renders correctly", () => {
  const render = renderer.create(<App />);

  expect(render).toMatchSnapshot();
});
