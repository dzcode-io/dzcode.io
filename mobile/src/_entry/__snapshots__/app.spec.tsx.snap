// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  />
</RNCSafeAreaProvider>
`;
